### PR TITLE
Cleanup examples

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: #151 

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes

Late summer 🧹 time!
 - move away from Kotlin Syntethic to bindings (taking path to Compose) and Kotlin extension in general
 - remove redundant xml layouts
 - remove redundant map lifecycle calls due to having Mapbox Lifecycle plugin enabled
 - fix warnings in activities
 - and some more minor things

### CODE REVIEW

**Review by Chaoba — APPROVED** [review]
LGTM

File: app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt [hunk 1]
```diff
@@ -165,31 +168,20 @@ class DebugModeActivity : AppCompatActivity() {
...
 
+  @SuppressLint("SetTextI18n")
   override fun onStart() {
     super.onStart()
-    mapView.onStart()
-    mapView.setOnFpsChangedListener {
+    binding.mapView.setOnFpsChangedListener {
       runOnUiThread {
-        fpsView.text = "${it.toInt()} FPS"
+        binding.fpsView.text = "${it.toInt()} FPS"
```
Comment (ank27) [inline app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt:176]: We can use string resources to remove `@SuppressLint("SetTextI18n")`.

**Review by ank27 — APPROVED** [review]

**Comment by ank27** [comment]
LGTM

File: app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt [hunk 2]
```diff
@@ -24,15 +23,16 @@ class DistanceExpressionActivity : AppCompatActivity() {
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_within_expression)
+    val binding = ActivitySimpleMapBinding.inflate(layoutInflater)
```
Comment (pengdev) [inline app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt:26]: Why not using `DistanceExpressionActivityBinding`?

File: app/src/main/res/layout/activity_within_expression.xml [hunk 3]
```diff
@@ -1,12 +0,0 @@
...
-    xmlns:mapbox="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent">
-
-    <com.mapbox.maps.MapView
-        android:id="@+id/mapView"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent" />
-
-</RelativeLayout>
```
Comment (pengdev) [inline app/src/main/res/layout/activity_within_expression.xml:12]: why remove?

File: app/src/main/res/layout/activity_style_geojson_layer_in_stack.xml [hunk 4]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<merge xmlns:android="http://schemas.android.com/apk/res/android">
-
-    <com.mapbox.maps.MapView
-        android:id="@+id/mapView"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent" />
-
-</merge>
```
Comment (pengdev) [inline app/src/main/res/layout/activity_style_geojson_layer_in_stack.xml:9]: why remove? imo we should have separate layout for different activities.

File: app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt (hunk 2 above)
Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt:26]: I've deleted all xml layouts that actually duplicate `activity_simple_map` and replaced.

File: app/src/main/res/layout/activity_lab_space_station_location.xml [hunk 5]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<merge xmlns:android="http://schemas.android.com/apk/res/android">
-
-    <com.mapbox.maps.MapView
-        android:id="@+id/mapView"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent" />
-
-</merge>
```
Comment (pengdev) [inline app/src/main/res/layout/activity_lab_space_station_location.xml:9]: same here.

File: app/src/main/res/layout/activity_dds_style_circles_categorically.xml [hunk 6]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<merge xmlns:android="http://schemas.android.com/apk/res/android">
-
-    <com.mapbox.maps.MapView
-        android:id="@+id/mapView"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent" />
-
-</merge>
```
Comment (pengdev) [inline app/src/main/res/layout/activity_dds_style_circles_categorically.xml:9]: same here

File: app/src/main/res/layout/activity_dds_draw_geojson_line.xml [hunk 7]
```diff
@@ -1,13 +0,0 @@
...
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".examples.linesandpolygons.DrawGeoJsonLineActivity">
-
-    <com.mapbox.maps.MapView
-        android:id="@+id/mapView"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent" />
-</merge>
```
Comment (pengdev) [inline app/src/main/res/layout/activity_dds_draw_geojson_line.xml:13]: same here

File: app/src/main/res/layout/activity_circle_layer_clustering.xml [hunk 8]
```diff
@@ -1,9 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<merge xmlns:android="http://schemas.android.com/apk/res/android">
-
-    <com.mapbox.maps.MapView
-        android:id="@+id/mapView"
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"/>
-
-</merge>
```
Comment (pengdev) [inline app/src/main/res/layout/activity_circle_layer_clustering.xml:9]: same here

File: app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt (hunk 2 above)
Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt:26]: imo we should have separate layout for different activities.

Comment (pengdev) [reply app/src/main/java/com/mapbox/maps/testapp/examples/DistanceExpressionActivity.kt:26]: It's confusing when reading the source code, imho it's better to make our examples more self contained so it's easier to follow by developers.

File: app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt (hunk 1 above)
Comment (kiryldz) [reply app/src/main/java/com/mapbox/maps/testapp/examples/DebugModeActivity.kt]: Yep!

**Review by pengdev — APPROVED** [review]

**Pull request merged**